lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -203,10 +231,23 @@ def create_instance(module, client, instance_name):
     zone = module.params.get('zone')
     blueprint_id = module.params.get('blueprint_id')
     bundle_id = module.params.get('bundle_id')
-    key_pair_name = module.params.get('key_pair_name')
     user_data = module.params.get('user_data')
     user_data = '' if user_data is None else user_data
 
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    if module.params.get('key_pair_name'):
+        key_pair_name = module.params.get('key_pair_name')
+    else:
+        key_pair_name = ''
+
+    if module.params.get('open_ports'):
+        open_ports = module.params.get('open_ports')
+    else:
+        open_ports = '[]'
```

Comment:
This should just be a list I think. Otherwise `if open_ports` is truthy and `for o in open_ports:` will look at the characters '[' and ']'. But rather than this, you can just add default=[] to the arg spec for this option.